src/sys/timer.cxx

```diff
@@ -5,20 +5,29 @@ using namespace std;
 
 Timer::Timer() {
   timing = getInfo("");
-  timing->started = MPI_Wtime();
-  timing->running = true;
+  if (timing->counter == 0) {
+    timing->started = MPI_Wtime();
+    timing->running = true;
+  }
+  timing->counter++;
 }
 
 Timer::Timer(const std::string &label) {
   timing = getInfo(label);
-  timing->started = MPI_Wtime();
-  timing->running = true;
+  if (timing->counter == 0) {
+    timing->started = MPI_Wtime();
+    timing->running = true;
+  }
+  timing->counter++;
 }
 
 Timer::~Timer() {
-  double finished = MPI_Wtime();
-  timing->running = false;
-  timing->time += finished - timing->started;
+  timing->counter--;
+  if (timing->counter == 0) {
+    double finished = MPI_Wtime();
```

Comment:
Do we need to store this or can we just use `MPI_Wtime` directly below?

Reply:
I was just adding the counters for this PR, as this method is changed again in 4.3